fe1-web/src/features/wallet/objects/Wallet.ts

```diff
@@ -1,45 +1,32 @@
-import { Hash, PopToken } from 'core/objects';
-import { getStore } from 'core/redux';
-import { LaoEventType } from 'features/events/objects/LaoEvent';
-import { makeEventByTypeSelector } from 'features/events/reducer';
+import { Lao } from 'features/lao/objects';
 import { RollCall } from 'features/rollCall/objects';
 
+import { RollCallToken } from './RollCallToken';
 import { generateToken } from './Token';
 
 /**
  * Recovers all PoP tokens associated with this wallet.
- *
- * @remarks
- * This is implemented by checking through all known Roll Calls of all known LAOs,
- * generating tokens for them and checking if the token is a verified attendee.
+ * @return Promise<RollCallToken[]>
  */
-export async function recoverWalletPoPTokens(): Promise<Record<string, Record<string, PopToken>>> {
-  const rollCallSelector = makeEventByTypeSelector<RollCall>(LaoEventType.ROLL_CALL);
-  // we're outside a component, we can't useSelector
-  const rollCalls = rollCallSelector(getStore().getState());
-
-  const tokens: Record<string, Record<string, PopToken>> = {};
-
-  let ops: Promise<any>[] = [];
-  for (const [laoId, laoRollCalls] of Object.entries(rollCalls)) {
-    tokens[laoId] = {};
-
-    // for each roll call
-    const newOps = Object.values(laoRollCalls).map(
-      // generate a token
-      (rc) =>
-        generateToken(new Hash(laoId), rc.id).then((token) => {
-          // if it's present in the roll call, add it
-          if (rc.containsToken(token)) {
-            tokens[laoId][rc.id.valueOf()] = token;
-          }
-        }),
-    );
-
-    ops = ops.concat(newOps);
-  }
-
-  await Promise.all(ops);
-
-  return tokens;
+export async function recoverWalletRollCallTokens(
```

Comment:
I have a strong concern here: it appears we've significantly modified the behaviour of the wallet and reduced the scope of its functionality.
Effectively, the wallet was (previously) able to display all the tokens from all the roll calls in all the LAOs.
Now, it's impossible to see the tokens from a LAO we're not connected to, despite this being in the [Digital Wallet specifications](https://docs.google.com/document/d/1bIEg31ckdsCnOT97uRgaDtkWHFbK2IJ4Zt5QyTNJxt8/edit?disco=AAAAYqLrbD4)

Reply:
It appeared to me that we did not need to see the tokens from every LAO, but only the ones in which we are connected because every action we could take with this token would be related to this particular LAO(or even this particular RollCall) ?

Reply:
It depends how you look at the Wallet. In principle, the wallet could be used offline / while disconnected from a LAO, though in relation with it. I believe that's one of the aims of the digital cash project.

If you think about it that way, it makes sense that you'd like to see (at least) all the current PoP tokens (from all LAOs) and possibly also historical (now invalid) PoP tokens.

Don't get me wrong, you're absolutely right that it might make sense to be able to restrict the view to a given LAO, and maybe a way to do that in the UI would be nice, I just wouldn't drop the ability to see everything else

Reply:
Yes, that is true, I think it makes more sense for now to keep this branch as is and I'll work around a way to be able to see every PopToken on another branch, for example by having an LAO selector, showing the LAO we're currently connected to and every other LAO and being able to select the one we are interested in.
Tell me if it makes sense to you

Reply:
It makes sense, given the already-wide scope of this PR.
Do try to address that issue in the next PR though, lest we lose functionality along the way :)